python/ray/projects/projects.py

```diff
@@ -0,0 +1,38 @@
+import json
+import jsonschema
+import os
+import yaml
+
+
+def find_root(dir):
+    """Find root directory of the ray project.
+
+    Args:
+        dir (str): Directory to start the search in.
+
+    Returns:
+        Path of the parent directory containing the .rayproject or
+        None if no such project is found.
+    """
+    prev, dir = None, os.path.abspath(dir)
+    while prev != dir:
+        if os.path.isdir(os.path.join(dir, ".rayproject")):
+            return dir
+        prev, dir = dir, os.path.abspath(os.path.join(dir, os.pardir))
+    return None
+
+
+def validate_project(project_file):
```

Comment:
I think more validation logic need to be added:
- Make sure the autoscaler yaml file exists. 
- Make sure all the path exists. (Dockerfile, requirements, etc)

Reply:
Fixed!